src/collective/documentgenerator/content/pod_template.py

```diff
@@ -119,27 +123,55 @@ class IMergeTemplatesRowSchema(zope.interface.Interface):
     )
 
 
+class PodFormatsValidator(validator.SimpleFieldValidator):
+    """ z3c.form validator class for international phone numbers """
+
+    def validate(self, selected_formats):
+        # thanks to widget, we get just-loaded file.filename.
+        widgets = self.view.widgets
+        current_filename = ""
+        for element in widgets.items():
+            if element[0] == 'odt_file':
+                current_filename = element[1].filename
+        if current_filename:
+            # avoid circular import.
+            from collective.documentgenerator.config import ODS_FORMATS
+            from collective.documentgenerator.config import ODT_FORMATS
+            from collective.documentgenerator.config import NEUTRAL_FORMATS
+            FORMATS_DICT = {'ods': ODS_FORMATS + NEUTRAL_FORMATS,
+                            'odt': ODT_FORMATS + NEUTRAL_FORMATS}
+            extension = current_filename.split('.')[-1]
+            authorise_element_list = FORMATS_DICT[extension]
+            authorise_extension_list = [elem[0] for elem in authorise_element_list]
+            if len(selected_formats) == 0:
+                raise Invalid(_(u"No format selected"))
+            for element in selected_formats:
+                if element not in authorise_extension_list:
+                    elem = self.get_invalid_error(element)
+                    error_message = _(u"element_not_valid",
```

Comment:
i18nmsgid is made to be returned into a template where translation is done, here, doing _() then translate could be replaced by :
from zope.i18n import translate
translate(msgid, domain, mapping, context=self.request)